src/tools/generators.ts

```diff
@@ -601,3 +608,275 @@ export async function generateAppIcons(option: `${Platforms}` | "all") {
+    type: "ios" | "android" | "web" | "all",
+    size: number,
+    expoRules: { name?: string; width: number; height: number; scale: number }[],
+  ) {
+    for (const expoRule of expoRules) {
+      const { name, width, height, scale } = expoRule
+
+      const outputFileName = [`splash-logo`, type, name].filter(Boolean).join("-") + ".png"
+      const outputFilePath = path(expoOutputDirPath, outputFileName)
+      const logoSize = size * scale
+      const verticalPadding = Math.ceil((height - logoSize) / 2)
+      const horizontalPadding = Math.ceil((width - logoSize) / 2)
+
+      try {
+        await sharp(inputFilePath)
+          .resize(logoSize, logoSize, { fit: "fill" })
+          .extend({
+            top: verticalPadding,
+            bottom: verticalPadding,
+            left: horizontalPadding,
+            right: horizontalPadding,
+            background: { r: 0, g: 0, b: 0, alpha: 0 },
+          })
+          .toFile(outputFilePath)
+
+        direction(`✅ assets/images/${outputFileName}`)
+        optionGenerationSuccesses.push(true)
+      } catch (error) {}
+    }
+  }
+
+  await generateForVanilla("ios", iosSize)
+  await generateForVanilla("android", androidSize)
+
+  heading(`Generating Expo splash screens (Android, iOS, and Web)...`)
+  if (isExpoOutputDirExists) {
+    await generateForExpo("ios", iosSize, [
+      { name: "mobile", width: 1284, height: 2778, scale: 3 },
+      { name: "tablet", width: 2048, height: 2732, scale: 2 },
+    ])
+    await generateForExpo("android", androidSize, [
+      { name: "universal", width: 1440, height: 2560, scale: 4 },
+    ])
+    await generateForExpo("web", 300, [{ width: 1920, height: 1080, scale: 1 }])
+    await generateForExpo("all", 180, [{ width: 1242, height: 2436, scale: 3 }])
+
+    // update app.json
+    const boilerplateDirPath = path(igniteCliRootDir(), "boilerplate")
+    const merge = require("deepmerge-json")
+    const sourceExpoConfig = require(path(boilerplateDirPath, "app.json"))?.expo
+    const outputAppConfig = require(path(cwd, "app.json"))
+
+    const updatedConfig = merge(outputAppConfig, {
+      expo: {
+        splash: {
+          backgroundColor,
+          image: sourceExpoConfig?.splash?.image,
+          resizeMode: sourceExpoConfig?.splash?.resizeMode,
+        },
+        android: {
+          splash: {
+            backgroundColor,
+            image: sourceExpoConfig?.android?.splash?.image,
+            resizeMode: sourceExpoConfig?.android?.splash?.resizeMode,
+          },
+        },
+        ios: {
+          splash: {
+            backgroundColor,
+            image: sourceExpoConfig?.ios?.splash?.image,
+            resizeMode: sourceExpoConfig?.ios?.splash?.resizeMode,
+          },
+        },
+        web: {
+          splash: {
+            backgroundColor,
+            image: sourceExpoConfig?.web?.splash?.image,
+            resizeMode: sourceExpoConfig?.web?.splash?.resizeMode,
+          },
+        },
+      },
+    })
```

Comment:
we're going to update the `app.json` with the new asset paths, colors and missing definitions